client/src/components/hero_location/heroLocationCtrl.js

```diff
@@ -19,9 +20,14 @@
       vm.selection = index;
     };
     vm.confirm = function() {
+      var queueHero = profileFactory.getProfile('username');
       //set location of hero to vm.locations[vm.selection]
-      heroFactory.setOrder(vm.locations[vm.selection]);
-      $state.go('hero_task');
+      ajaxFactory.setHeroLocation(queueHero, vm.locations[vm.selection])
+        //will be executed if status code is 200-299,
+        .then(function successCallback(response) {
+          heroFactory.setOrder(vm.locations[vm.selection]);

```

Comment:
line 28, setOrder is setting a yelp object in the heroFactoy... these objects are very different. the yelp object has keys like "name, address, city, state, zip, etc.." heroFactory object has keys like "transactionId, queueHero, requester, item, vendor, etc.) you should reconcile all the valuable data and put what is necessary into heroFactory.

Feel free to add keys to the heroFactory data object if you believe they are necessary for future views.

Also, does yelp have a uniqueID to identify a vendor? I would guess yelp's venue.name is not a unique identifier.